### DISCUSSION TITLE
🔦 Feature: a FAQ clarifying the nocodb license

### DISCUSSION THREAD

**Discussion by gautierrog** [nocodb/nocodb#6741] [General]
### Please confirm if feature request does NOT exists already ?

- [X] I confirm there is no existing issue for this

### Describe the usecase for the feature

I was looking for information about nocodb licensing and even though the project is GNU AGPL I am a bit confused about what we can / cannot do with it.

Basically, I just want to know if I can host nocodb and sell services to clients freely or do I have to pay a license in order to do it.
I saw this pricing page https://nocodb.com/pricing and I guess that the fee are about the hosted version of nocodb. Am I right?

### Suggested Solution

Have FAQ page with can do / cannot do with the product in a license point of view

### Additional Context

just looking for information

**Comment by orthoxerox** [comment]
AGPL says you can sell services freely, but all users must be granted access to the source code of the version you are running with the same rights. This means that if you run NocoDB unmodified, you can show them a link to this repository. If you modify NocoDB (for example, add a new field type or an additional authentication method), then you must tell your users they can ask for the source code and you must give it to them under the same license, so your users are free to run their own instances with your changes.

**Comment by cedstrom** [comment]
How does this square with Noco themselves withholding sources to some of the enterprise features?  For example, the BulkEdit component was removed in #6222 and I was told on Discord that this is an enterprise only feature now.  This also applies to the SAML/SSO features. Can I request the source code to these EE exclusives?

**Reply by orthoxerox** [reply]
I am not an intellectual property expert, but EE is very likely not licensed under AGPL. Nocodb, Inc is free to distribute their own software under different licenses, as everyone else who wants to contribute must sign a CLA with Nocodb, Inc first.

This means they can't withhold the source of the version they released under AGPL, but are perfectly allowed to withhold the source of the version they extended with enterprise features.

**Comment by Impaler67** [comment]
### 2026.04 UPD
Whoever face this post, NocoDB is no longer an open source product. Its licensing approach has been changed to Fair Code (Sustainable Use License). Read [the commit message](https://github.com/nocodb/nocodb/pull/12801)
